### PR TITLE
fix(source): drop postgres replication slot in `drop table`

### PR DESCRIPTION
**Pull request by WillyKidd**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- for each postgres sources, a new replication slot is created if the user does not specify `slot.name`
- too many useless replication slots is bad
- therefore we delete the replication slot created by rw (slots with the name `rw_cdc_*`) during `drop table`
  - a new rpc for the connector node `drop replication slot`, which tells the connector to drop the replication slot after it stops
  - meta cannot directly tell the connector node to drop replication, because meta cannot know exactly when the connector will stop gracefully..(there might be a delay before connector notices that the `getEventStream` rpc has broken)

related:
- https://github.com/risingwavelabs/risingwave/issues/10027#issuecomment-1590773176
- https://github.com/risingwavelabs/risingwave-docs/pull/669#issuecomment-1478527159

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

## Documentation

- [x] My PR does NOT contain user-facing changes.

### CODE REVIEW

File: src/meta/src/rpc/ddl_controller.rs [hunk 1]
```diff
@@ -614,6 +620,45 @@ where
             self.source_manager
                 .unregister_sources(vec![source_id])
                 .await;
+            // drop replication slot for pg sources
+            if source.get_properties().get("connector") == Some(&POSTGRES_CDC_CONNECTOR.to_string())
```
Comment (yezizp2012) [inline src/meta/src/rpc/ddl_controller.rs:624]: Just like what I commented in https://github.com/risingwavelabs/risingwave/pull/10455#discussion_r1236458142, do we have any other mechanisms to make sure that the slot will be cleaned if any failover occurs? Cc. @StrikeW

**Review by yezizp2012 — COMMENTED** [review]
Meta part LGTM.

File: src/meta/src/rpc/ddl_controller.rs (hunk 1 above)
Comment (StrikeW) [reply src/meta/src/rpc/ddl_controller.rs:624]: > do we have any other mechanisms to make sure that the slot will be cleaned if any failover occurs?

I think not right now. Meta completely relies on the etcd transaction to provide atomicity, I am considering whether we can also leverage etcd to provide atomicity for external resources cleanup.

File: java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzSourceHandler.java [hunk 2]
```diff
@@ -29,9 +34,12 @@
 public class DbzSourceHandler implements SourceHandler {
     static final Logger LOG = LoggerFactory.getLogger(DbzSourceHandler.class);
 
+    private Map<Long, String> replicationSlotMap;
```
Comment (StrikeW) [inline java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzSourceHandler.java:37]: I think we can use a singleton class to store the slot name for each source, then you don't need to pass the replicationSlotMap all the way of the call stack. Just access the singleton class is enough, which can make the code cleaner.

**Pull request closed without merging**